client/blocks/login/two-factor-authentication/push-notification-approval-poller.jsx

```diff
@@ -1,42 +1,96 @@
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
-import { startPollAppPushAuth, stopPollAppPushAuth } from 'calypso/state/login/actions';
-import { getTwoFactorPushPollSuccess } from 'calypso/state/login/selectors';
-
-class PushNotificationApprovalPoller extends Component {
-	static propTypes = {
-		onSuccess: PropTypes.func.isRequired,
-		pushSuccess: PropTypes.bool.isRequired,
-		startPollAppPushAuth: PropTypes.func.isRequired,
-		stopPollAppPushAuth: PropTypes.func.isRequired,
-	};
-
-	componentDidMount() {
-		this.props.startPollAppPushAuth();
-	}
+import config from '@automattic/calypso-config';
+import { useEffect, useRef } from 'react';
+import { useDispatch } from 'react-redux';
+import { localizeUrl } from 'calypso/lib/i18n-utils';
+import { updateNonce } from 'calypso/state/login/actions';
+import { remoteLoginUser } from 'calypso/state/login/actions/remote-login-user';
+import {
+	getTwoFactorAuthNonce,
+	getTwoFactorPushToken,
+	getTwoFactorUserId,
+} from 'calypso/state/login/selectors';
 
-	componentWillUnmount() {
-		this.props.stopPollAppPushAuth();
-	}
+const POLL_APP_PUSH_INTERVAL = 5 * 1000;
 
-	componentDidUpdate( prevProps ) {
-		if ( ! prevProps.pushSuccess && this.props.pushSuccess ) {
-			this.props.onSuccess();
-		}
-	}
+async function request( state ) {
+	const url = localizeUrl(
+		'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint'
+	);
+	const body = new URLSearchParams( {
+		user_id: getTwoFactorUserId( state ),
+		auth_type: 'push',
+		remember_me: true,
+		two_step_nonce: getTwoFactorAuthNonce( state, 'push' ),
+		two_step_push_token: getTwoFactorPushToken( state ),
+		client_id: config( 'wpcom_signup_id' ),
+		client_secret: config( 'wpcom_signup_key' ),
+	} );
 
-	render() {
-		return null;
-	}
+	const response = await fetch( url, {
+		method: 'POST',
+		credentials: 'include',
+		body,
+	} );
+
+	return await response.json();
 }
 
-export default connect(
-	( state ) => ( {
-		pushSuccess: getTwoFactorPushPollSuccess( state ),
-	} ),
-	{
-		startPollAppPushAuth,
-		stopPollAppPushAuth,
+const poll = ( signal ) => async ( dispatch, getState ) => {
+	let aborted = false;
+	signal.addEventListener( 'abort', () => {
+		aborted = true;
+	} );
+
+	while ( true ) {
```

Comment:
Aha! So that's why you needed #58344 for 😅

Reply:
This is the second time this month where I got complaints about `while ( true )`. The first one was `wpcom-xhr-request` in #57719. So I decided to do something about them 🙂